ballerina/modules/apex/client.bal

```diff
@@ -0,0 +1,97 @@
+    # [this guide](https://help.salesforce.com/articleView?id=remoteaccess_authenticate_overview.htm). 
+    #
+    # + salesforceConfig - Salesforce Connector configuration
+    # + return - `sfdc:Error` on failure of initialization or else `()`
+    public isolated function init(ConnectionConfig config) returns error? {
+        http:Client|http:ClientError|error httpClientResult;
+        http:ClientConfiguration httpClientConfig = check config:constructHTTPClientConfig(config);
+        httpClientResult = trap new (config.baseUrl, httpClientConfig);
+
+        if httpClientResult is http:Client {
+            self.salesforceClient = httpClientResult;
+        } else {
+            return error(INVALID_CLIENT_CONFIG);
+        }
+    }
+
+    # Access Salesforce APEX resource.
+    #
+    # + urlPath - URI path
+    # + methodType - HTTP method type
+    # + payload - Payload
+    # + returnType - The payload type, which is expected to be returned after data binding
+    # + return - `string|int|record{}` type if successful or else `error`
+    isolated remote function apexRestExecute(string urlPath, http:Method methodType,
+            record {} payload = {}, typedesc<record {}|string|int?> returnType = <>)
+            returns returnType|error = @java:Method {
+        'class: "io.ballerinax.salesforce.ReadOperationExecutor",
+        name: "apexRestExecute"
+    } external;
+
+    private isolated function processApexExecute(typedesc<record {}|string|int?> returnType, string urlPath, http:Method methodType, record {} payload) returns record {}|string|int|error? {
+        string path = utils:prepareUrl([APEX_BASE_PATH, urlPath]);
+        http:Response response = new;
+        match methodType {
+            "GET" => {
+                response = check self.salesforceClient->get(path);
+            }
+            "POST" => {
+                response = check self.salesforceClient->post(path, payload);
+            }
+            "DELETE" => {
+                response = check self.salesforceClient->delete(path);
+            }
+            "PUT" => {
+                response = check self.salesforceClient->put(path, payload);
+            }
+            "PATCH" => {
+                response = check self.salesforceClient->patch(path, payload);
+            }
+            _ => {
+                return error("Invalid Method");
+            }
+        }
+        if response.statusCode == 200 || response.statusCode == 201 {
+            if response.getContentType() == "" {
+                return;
+            }
+            json responsePayload = check response.getJsonPayload();
+            return check responsePayload.cloneWithType(returnType);
+        } else {
+            json responsePayload = check response.getJsonPayload();
+            return error("Error occurred while executing the apex request. ",
+                httpCode = response.statusCode, details = responsePayload);
```

Comment:
Do we need to typecast this? Shouldnt this be handled in the http call itself?

Reply:
Mentioned in https://github.com/ballerina-platform/ballerina-library/issues/6697